control_plane/attachment_service/src/service.rs

```diff
@@ -720,6 +815,12 @@ impl Service {
 
         let (startup_completion, startup_complete) = utils::completion::channel();
 
+        let cancel = CancellationToken::new();
+        let heartbeater_handler = HeartbeaterHandler::new(
+            config.jwt_token.clone(),
+            Duration::from_secs(10),
```

Comment:
Can you pull this duration up to a constant at the top of the file?

(I'm also guilty of putting durations inline, need to be be tighter on this)

Reply:
Actually, this probably needs to be externally configurable.  We don't have a config file yet, but this is important enough to make a CLI flag (there's a Config object you can thread it through).

The default should probably be something quite high, initially high enough to cover a node restart, since marking a node offline (and thereby moving attachments) is quite a heavy thing to do.  We can tune it down to something smaller as we build confidence.

Reply:
Speaking of configuration, as a followup it would be good to think about a mechanism for a human to prevent a node flapping.  e.g. to say "keep this offline even if it responds to heartbeats" and/or "keep this online even if it misses heartbeats".

I was thinking of a "Paused" scheduling state for similar scheduling concerns, maybe that could do double-duty here: a node with the paused state should have no automatic behaviors at all, including marking offline/active.  The more general thing would be to introduce  a persistent AvailabilityBehavior or something, but that feels quite heavy weight and we really just need something good enough to stop the system thrashing around when we're doing something during an incident.

Reply:
Made configurable in https://github.com/neondatabase/neon/pull/7092/commits/835af9d447c43f617e96dc08a3a21255e8361e77